org.eclipse.ui.console/src/org/eclipse/ui/internal/console/AnsiConsoleColorPalette.java

```diff
@@ -0,0 +1,304 @@
+package org.eclipse.ui.internal.console;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.RGB;
+
+public class AnsiConsoleColorPalette {
```

Comment:
In the future (not necessary now), we should build a system palette relying on system colors as made available via `Display.getSystemColor(SWT.COLOR_RED)` and so on.